hazelcast/src/test/java/com/hazelcast/client/tpc/ClientAsyncSocketReaderTest.java

```diff
@@ -0,0 +1,259 @@
+            ClientTestUtil.writeClientMessage(os, largeRequest);
+            ClientMessage largeResponse = ClientTestUtil.readResponse(is);
+            assertEquals(ExperimentalTpcAuthenticationCodec.RESPONSE_MESSAGE_TYPE, largeResponse.getMessageType());
+        }
+    }
+
+    @Test
+    public void testMessagesBeforeAuth() throws IOException {
+        HazelcastInstance server = newServer();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
+             OutputStream os = socket.getOutputStream();
+             InputStream is = socket.getInputStream()) {
+
+            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+            ClientMessage request = ClientPingCodec.encodeRequest();
+            ClientTestUtil.writeClientMessage(os, request);
+            assertSocketClosed(is);
+        }
+    }
+
+    @Test
+    public void testAuth_withNotFoundUuid() throws IOException {
+        HazelcastInstance server = newServer();
+        newClient();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
+             OutputStream os = socket.getOutputStream();
+             InputStream is = socket.getInputStream()) {
+
+            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+            ClientMessage request = encodeAuthRequest(UUID.randomUUID(), getTpcToken(server));
+            ClientTestUtil.writeClientMessage(os, request);
+            assertSocketClosed(is);
+        }
+    }
+
+    @Test
+    public void testAuth_withMismatchedToken() throws IOException {
+        HazelcastInstance server = newServer();
+        HazelcastInstance client = newClient();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
+             OutputStream os = socket.getOutputStream();
+             InputStream is = socket.getInputStream()) {
+
+            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+            ClientMessage request = encodeAuthRequest(getUuid(client), new byte[0]);
+            ClientTestUtil.writeClientMessage(os, request);
+            assertSocketClosed(is);
+        }
+    }
+
+    @Test
+    public void testAuth_withMismatchedToken_forTpcDisabledClient() throws IOException {
+        HazelcastInstance server = newServer();
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
+             OutputStream os = socket.getOutputStream();
+             InputStream is = socket.getInputStream()) {
+
+            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+            ClientMessage request = encodeAuthRequest(getUuid(client), new byte[0]);
+            ClientTestUtil.writeClientMessage(os, request);
+            assertSocketClosed(is);
+        }
+    }
+
```

Comment:
We can add this one as well maybe: 

```java
@Test
    public void testAuth_withNotFoundUuid_forTpcDisabledClient() throws IOException {
        HazelcastInstance server = newServer();
        HazelcastClient.newHazelcastClient();

        int port = getTpcPort(server);
        try (Socket socket = new Socket("127.0.0.1", port);
             OutputStream os = socket.getOutputStream();
             InputStream is = socket.getInputStream()) {

            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
            ClientMessage request = encodeAuthRequest(UUID.randomUUID(), getTpcToken(server));
            ClientTestUtil.writeClientMessage(os, request);
            assertSocketClosed(is);
        }
    }
```

Reply:
I don't think it is necessary. When the UUID is not found, whether or not there is a client, or it is a TPC-enabled or disabled client doesn't really matter. We just can't find a UUID for the given client